go/vt/binlog/binlog_streamer_rbr_test.go

```diff
@@ -270,19 +270,8 @@ func TestStreamerParseRBREvents(t *testing.T) {
 
 	go sendTestEvents(events, input)
 	_, err := bls.parseEvents(context.Background(), events, errs)
-	if err != ErrServerEOF {
-		t.Errorf("unexpected error: %v", err)
-	}
-
-	if !reflect.DeepEqual(got, want) {
-		t.Errorf("binlogConnStreamer.parseEvents(): got:\n%+v\nwant:\n%+v", got, want)
-		for i, fbt := range got {
-			t.Errorf("Got (%v)=%v", i, fbt.statements)
-		}
-		for i, fbt := range want {
-			t.Errorf("Want(%v)=%v", i, fbt.statements)
-		}
-	}
+	assert.EqualError(t, err, ErrServerEOF.Error(), "unexpected error")
```

Comment:
Please revert the order: `ErrServerEOF.Error()` is the expected value, `err` is the tested value. Should be:
```suggestion
	assert.EqualError(t, ErrServerEOF.Error(), err, "unexpected error")
```

Reply:
After checking the official package [link](https://pkg.go.dev/github.com/stretchr/testify/assert) the format is described as:

`assert.EqualError(t *testing.T, theError error, errString string, msgAndArgs ...interface{}) bool`

EqualError asserts that a function returned an error (i.e., not nil) and that it is equal to the provided error.
```
actualObj, err := SomeFunction()
assert.EqualError(t, err, expectedErrorString)
```
I have done a similar process here to follow the rules. I also followed your suggestion of reverting the order, but I encountered two errors:

ErrServerEOF.Error()
```
cannot use ErrServerEOF.Error() (value of type string) as error value in argument to assert.EqualError: string does not implement error (missing method Error) compiler[InvalidIfaceAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#InvalidIfaceAssign)
var ErrServerEOF error
```
err
```
cannot use err (variable of type error) as string value in argument to assert.EqualError compiler[IncompatibleAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#IncompatibleAssign)
var err error
```

Reply:
My bad! Yes, you're right about `EqualError`.